### PR TITLE
Make it clearer you're creating a new question when saving

### PR DESCRIPTION
**Pull request by mazameli**
## When you're saving a new question, like when starting from a model or table 

image

## When you started from a saved question and modified it (this state is unchanged from master)

image

### CODE REVIEW

**Review by brunobergher — COMMENTED** [review]
The copy change seems good, the nested ternaries not sure. Plus an indentation thing on line 188. I'll let an actual engineer comment on those.

File: frontend/src/metabase/containers/SaveQuestionModal.jsx [hunk 1]
```diff
@@ -104,17 +104,21 @@ export default class SaveQuestionModal extends Component {
...
     const showSaveType =
       !card.id &&
       !!originalCard &&
       !originalCard.dataset &&
       originalCard.can_write;
 
+    const singleStepTitle =
+      questionType === "question"
+        ? showSaveType
+          ? t`Save question`
```
Comment (mazameli) [inline frontend/src/metabase/containers/SaveQuestionModal.jsx:116]: Not sure how to do this via if-elses when the conditional is within the instantiation of a const like this. I'm sure there's a better pattern?

Comment (ranquild) [reply frontend/src/metabase/containers/SaveQuestionModal.jsx:116]: We usually extract it to a function with multiple returns to keep using const everywhere

Comment (mazameli) [reply frontend/src/metabase/containers/SaveQuestionModal.jsx:116]: @ranquild KK, thanks. I'm sure I'm doing this wrong 'cause I hit a server error when I try to save now, but am I headed in the right general direction with this? https://gist.github.com/mazameli/6ae5b8d7f4e06119d79014f43019c1a1

Comment (ranquild) [reply frontend/src/metabase/containers/SaveQuestionModal.jsx:116]: @mazameli Something like this should work. I'd put this function outside of the component and call it here.

```
const getSingleStepTitle = (questionType, showSaveType) => {
  if (questionType === "model") {
    return t`Save model`;
  } else if (showSaveType) {
    return t`Save question`;
  } else {
    return t`Save new question`;
  }
};

```

**Comment by mazameli** [comment]
@ranquild okay, this should be ready for an actual review now if you have a moment. 😅

**Review by ranquild — APPROVED** [review]

**Comment by mazameli** [comment]
`native-mongo.cy.spec.js` keeps failing on me. cy.type() seems to be calling 2 elements.

```
  1) scenarios > question > native > mongo
       can save a native MongoDB query:
     CypressError: `cy.type()` can only be called on a single element. Your subject contained 2 elements.
```

**Comment by nemanjaglumac** [comment]
> `native-mongo.cy.spec.js` keeps failing on me. cy.type() seems to be calling 2 elements.
> 
> ```
>   1) scenarios > question > native > mongo
>        can save a native MongoDB query:
>      CypressError: `cy.type()` can only be called on a single element. Your subject contained 2 elements.
> ```

The problem you encountered is a flake OR a bug in frontend, but this test would fail even if it wasn't for that.

I like how Deploysentinel makes this easy to debug 🎉 
Didn't even have to spin up Cypress locally and/or to have Mongo DB running.

Let's take a look at the [initial attempt](https://www.deploysentinel.com/ci/runs/6378271c5d628751f31b71f6).
image

We changed the string from "Save question" to "Save new question" but the test hasn't been updated. This would **always** fail.

Both retry 1 and retry 2 failed at the very beginning of the test (that's why you never even encountered the real error from the initial step. I'm not entirely sure why our frontend renders two sql editors in DOM at one point, but it seems that it does so. Cypress is much faster than a human being, so it unfortunately caught the DOM in that state with two elements.

If we inspect the DOM (which is another great thing about DS) and then go to the console, we can see that there are indeed two `.ace_content` elements.
image

I'll fix the root issue, but this flakiness will creep up from time to time until we solve it in frontend.

**Pull request merged**